### DISCUSSION TITLE
Add Wi-Fi to device tracker location SourceType

### DISCUSSION THREAD

**Discussion by Oliv4945** [home-assistant/architecture#1042] [Entity Models] [answered]
# Context

Current device tracker allows to deliver the type of the location thanks to the parameter `SourceType` , which current values  are `gps`, `router`, `bluetooth` and `bluetooth_le`.   
Some trackers are able to scan surrounding Wi-Fi access points MAC address and resolve a GPS position against databases like [Mozilla Location Services](https://ichnaea.readthedocs.io/en/latest/api/geolocate.html#wifi-access-point-fields) or [Google](https://developers.google.com/maps/documentation/geolocation/requests-geolocation#wifi_access_point_object). To be accepted by Home Assistant we are currently required to use the source `gps` when source location is Wi-Fi, but accurately knowing the source type allows for more automation possibilities.

# Proposal

Add a new `wifi` source type:

```Python
    ROUTER = "router"
    BLUETOOTH = "bluetooth"
    BLUETOOTH_LE = "bluetooth_le"
    WIFI = "wifi"
```

# Consequences

- `device_tracker` and `person` needs to be updated. 
- Documentation needs to be updated. Proposed PR [here](https://github.com/home-assistant/home-assistant.io/pull/31546)
- Developer documentation needs to be updated. Proposed PR [here](https://github.com/home-assistant/developers.home-assistant/pull/2084)

As I initially did not understood that this would require a discussion in this repo I already created this [PR](https://github.com/home-assistant/core/pull/111111)

Thanks

**Comment by frenck** [comment]
Thanks for the arch proposal.

As with any arch discussion/proposal, the question is: Which core integrations will use this?
I think this request/suggestion is reasonable, but we generally only want to adjust our core models if there is a use-case for our integration.

../Frenck

**Reply by Oliv4945** [reply]
Hello Frenck,
There is no core integration yet, this is a chicken and egg problem :) In order to keep the PRs small I am starting with this before submitting the tracker PR

**Reply by emontnemery** [reply]
@Oliv4945 the question is which existing core integrations could make use of the proposal if it's accepted.

**Comment by emontnemery** [comment]
@Oliv4945 Could you please elaborate a bit more on how this would be used?

**Reply by Oliv4945** [reply]
Hello @emontnemery 

Use case:  
Knowing the location type allows to create different kind of automation depending on it, mainly because of the precision. But I am also using this to detect if I am indoor(Wi-Fi) or outdoor(GPS), with a really low false positive rate.

Existing core integrations benefiting from this change could be:  
- Owntracks
- Traccar
- iOS and Android applications

**Comment by frenck** [comment, marked as answer]
We have discussed this proposal (and the PRs) in our core team meeting, and concluded that we don't want to add this feature at this moment in time. We think that the use of this, is too shallow to add at this point.

Additionally, the services by Mozilla are sunset as well.

Nevertheless, thanks for bring up a proposal 👍 

../Frenck
